src/ray/core_worker/core_worker.cc

```diff
@@ -2075,21 +2075,33 @@ void CoreWorker::HandleGetObjectStatus(const rpc::GetObjectStatusRequest &reques
     send_reply_callback(Status::OK(), nullptr, nullptr);
   } else {
     RAY_CHECK(owner_address.worker_id() == request.owner_worker_id());
+    bool is_freed = reference_counter_->IsPlasmaObjectFreed(object_id);
 
-    if (reference_counter_->IsPlasmaObjectFreed(object_id)) {
-      reply->set_status(rpc::GetObjectStatusReply::FREED);
-    } else {
-      reply->set_status(rpc::GetObjectStatusReply::CREATED);
-    }
     // Send the reply once the value has become available. The value is
     // guaranteed to become available eventually because we own the object and
     // its ref count is > 0.
-    // TODO(swang): We could probably just send the object value if it is small
-    // enough and we have it local.
-    memory_store_->GetAsync(object_id,
-                            [send_reply_callback](std::shared_ptr<RayObject> obj) {
-                              send_reply_callback(Status::OK(), nullptr, nullptr);
-                            });
+    memory_store_->GetAsync(object_id, [reply, send_reply_callback,
+                                        is_freed](std::shared_ptr<RayObject> obj) {
+      if (is_freed) {
+        reply->set_status(rpc::GetObjectStatusReply::FREED);
+      } else if (obj->HasData() &&
```

Comment:
Assuming here that if the object is small enough for us to want to inline it in the `GetObjectStatus` response, then the object is in the in-memory store and the `RayObject` has actual data.

Reply:
I think we could just always send the object data, even if it's stored in plasma. The `RayObject` should contain the `OBJECT_IN_PLASMA` error in that case, so the receiver will retry getting the object from the object store once it receives the inlined value in its in-memory store.

Reply:
Ah, that's a great point. So I would assume that the `max_get_status_object_size` config check isn't necessary, then, nor is the `INLINED` object status, since the `GetObjectStatus` caller can just directly construct the `RayObject` from the proto message in either case and pass that to `in_memory_store_.Put()`.